src/Amazon.Extensions.CognitoAuthentication/CognitoAuthenticationClasses.cs

```diff
@@ -26,6 +26,23 @@ namespace Amazon.Extensions.CognitoAuthentication
     /// </summary>
     public class AuthFlowResponse
     {
+        /// <summary>
+        /// Constructs an AuthFlowResponse object
+        /// </summary>
+        /// <param name="sessionId">The authentication workflow session id.</param>
+        /// <param name="authenticationResult">The result of the Authentication workflow</param>
+        /// <param name="challengeName">The challenge name if any.</param>
+        /// <param name="challengeParameters">The challenge parameters if any.</param>
+        /// <param name="clientMetadata">The client metadata.</param>
+        public AuthFlowResponse(string sessionId, AuthenticationResultType authenticationResult, ChallengeNameType challengeName, IDictionary<string, string> challengeParameters, IDictionary<string, string> clientMetadata)
+        {
+            SessionID = sessionId;
```

Comment:
remove "internal set" from all property declarations. They are readonly now, and should prevent external write access.